File: baekjoon_all/11000+/boj_11161.cpp
// Solve 2023-12-30

#include <bits/stdc++.h>
using namespace std;

#define FASTIO ios_base::sync_with_stdio(false);cin.tie(NULL); // boj_15552.cpp
#define SETPRECISION(n) cout << fixed;cout.precision(n); // boj_1008.cpp
#define SIZE(v) (int)v.size()
#define ALL(v) v.begin(),v.end()
using ll = long long;

int main() {
    FASTIO;

    int t;
    cin >> t;

    for (int ti = 0; ti < t; ti++) {
        int m;
        cin >> m;

        int cur = 0;
        int ans = 0;

        for (int mi = 0; mi < m; mi++) {
            int p1, p2;
            cin >> p1 >> p2;

            cur += p1 - p2;
            ans = min(ans, cur);
        }

        cout << -ans << '\n';
    }

    return 0;
}

File: baekjoon_all/11000+/boj_11161.py
# Solve 2023-12-30

import sys

input = lambda : sys.stdin.readline().rstrip()

for _ in range(int(input())):
    ans = cur = 0

    for _ in range(int(input())):
        p1, p2 = map(int, input().split())
        cur += p1 - p2
        ans = min(ans, cur)

    print(-ans)
